src/main/ruby/core/exception.rb

```diff
@@ -390,37 +401,44 @@ class KeyError < IndexError
 
 class SignalException < Exception
 
+  alias :signm :message
   attr_reader :signo
-  attr_reader :signm
-
-  def initialize(signo = nil, signm = nil)
-    # MRI overrides this behavior just for SignalException itself
-    # but not for anything that inherits from it, therefore we
-    # need this ugly check to make sure it works as intended.
-    return super(signo) unless self.class == SignalException
-    if signo.is_a? Integer
-      unless @signm = Signal::Numbers[signo]
-        raise ArgumentError, "invalid signal number #{signo}"
+
+  def initialize(sig, message = undefined)
+    signo = Rubinius::Type.rb_check_to_integer(sig, :to_int)
+    if signo.nil?
+      raise ArgumentError, 'wrong number of arguments (given 2, expected 1)' unless undefined.equal?(message)
+      if sig.is_a?(Symbol)
+        sig = sig.to_s
+      else
+        sig = StringValue(sig)
       end
-      @signo = signo
-      @signm = signm || "SIG#{@signm}"
-    elsif signo
-      if signm
-        raise ArgumentError, 'wrong number of arguments (2 for 1)'
+      signm = sig
+      if signm.start_with?('SIG')
+        signm = signm[3..-1]
       end
-      signm = signo
-      if signo.kind_of?(Symbol)
-        signm = signm.to_s
-      else
-        signm = StringValue(signm)
+      signo = Signal::Names[signm]
+      if signo.nil?
+        raise ArgumentError, "invalid signal name SIG#{sig}"
       end
-      signm = signm[3..-1] if signm.start_with? 'SIG'
-      unless @signo = Signal::Names[signm]
-        raise ArgumentError, "invalid signal name #{signm}"
+      sig = 'SIG%s' % signm
+    else
+      if signo < 0 || !Signal::Numbers.has_key?(signo)
+        raise ArgumentError, "invalid signal number (#{signo})"
       end
-      @signm = "SIG#{signm}"
+      sig = if undefined.equal?(message)
+              s = Signal::Numbers[signo]
+              if s.nil?
+                'SIG%d' % signo
```

Comment:
This case is no longer possible, is it?
Since we check that signo is in Signal::Numbers above.

Reply:
I found how this is possible, for example on macOS:
```
irb(main):002:0> SignalException.new(32)
=> #<SignalException: SIG32>
```
I think `SignalException.new(0)` on Linux might do this too.

Reply:
I see, 32 also shows the same on Linux.
This sounds very much of a corner case, but given it seems easy to implement, it sounds good.
It would be good to have a spec or test for it though, unless there is already a spec/test covering this.